sui_core/src/authority.rs

```diff
@@ -806,3 +805,30 @@ impl ModuleResolver for AuthorityState {
         self._database.get_module(module_id)
     }
 }
+
+#[async_trait]
+impl ExecutionState for AuthorityState {
+    type Transaction = CertifiedTransaction;
+    type Error = SuiError;
+
+    async fn handle_consensus_transaction(
+        &self,
+        execution_indices: ExecutionIndices,
+        transaction: Self::Transaction,
+    ) -> Result<(), Self::Error> {
+        self.handle_consensus_certificate(transaction, execution_indices)
+            .await
+    }
+
+    fn ask_consensus_write_lock(&self) -> bool {
+        self.consensus_guardrail.fetch_add(1, Ordering::SeqCst) == 0
+    }
+
+    fn release_consensus_write_lock(&self) {
+        self.consensus_guardrail.fetch_sub(0, Ordering::SeqCst);
+    }
+
+    async fn load_execution_indices(&self) -> Result<ExecutionIndices, Self::Error> {
+        self._database.last_consensus_index()
```

Comment:
can we rename `_database` to `database` since its used now?

Reply:
That is also an easy fix but it seems that many currently opened PR will have to rebase if I do it now. So I made an issue that I will quickly fix separately https://github.com/MystenLabs/sui/issues/1497